### PR TITLE
Packing for libEigen dynamic matrices

### LINKED ISSUE #3662: Add `Packing` specializations for Eigen matrix types
Add specializations for dynamic Eigen matrices (and vectors). This will make comm code easier for e.g. tuples of matrices and vectors.

### PR DESCRIPTION
**Pull request by dschwen**
Closes #3662

### CODE REVIEW

**Comment by roystgnr** [comment]
Added some non-default recipes I'm skeptical about, but at first glance, modulo a few possible missing ifndefs, this is looking good.

File: include/parallel/parallel_eigen.h [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+// libMesh includes
+#include "libmesh/libmesh_config.h"
+#include "libmesh/libmesh_common.h"
+#include "libmesh/int_range.h"
+
+// TIMPI includes
+#include "timpi/packing.h"
+
+// libEigen includes
```
Comment (roystgnr) [inline include/parallel/parallel_eigen.h:29]: Pretty much everything from here down ought to be wrapped in `#ifdef LIBMESH_HAVE_EIGEN`

**Review by roystgnr — CHANGES_REQUESTED** [review]
This looks like it's good to go as soon as the --disable-eigen case is fixed.

**Comment by roystgnr** [comment]
Ah, hell.  I should have merged yours before mine.  Mind fixing up or adding a new rebootstrap commit?

**Comment by dschwen** [comment]
sure, no problem

**Comment by dschwen** [comment]
I manually fixed the conflict

**Comment by roystgnr** [comment]
Just the network issues with the python test again on `Test MOOSE mac`; this is good to go.

**Pull request merged**